src/savegame_json.cpp

```diff
@@ -3435,9 +3435,20 @@ void submap::store( JsonOut &jsout ) const
         jsout.end_array();
     }
     jsout.end_array();
-    // Output the computer
-    if( comp != nullptr ) {
-        jsout.member( "computers", comp->save_data() );
+
+    if( legacy_computer ) {
+        // it's possible that no access to computers has been made and legacy_computer
+        // is not cleared
+        jsout.member( "computers", legacy_computer->save_data() );
+    } else if( !computers.empty() ) {
+        jsout.member( "computers" );
+        jsout.start_array();
+        for( auto &elem : computers ) {
```

Comment:
Doesnt this need a call to `start_object` and `end_object` for each loop iteration? You are constructing a array of objects here, correct?

Reply:
Not an array of objects, but an array of consecutive integers and strings, as it was done in nearby code.